util/microservice-sdk/js/lib/kafka/Producer.js

```diff
@@ -110,19 +120,23 @@ class Producer {
     const readyPromise = new Promise((resolve, reject) => {
       const timeoutTrigger = setTimeout(() => {
         logger.error('Failed to connect the producer.', TAG);
-        reject(new Error('timed out'));
+        return reject(new Error('timed out'));
       }, this.config['producer.connect.timeout.ms']);
 
       this.producer.on('ready', () => {
         logger.debug('Producer is ready.', TAG);
         clearTimeout(timeoutTrigger);
         this.isReady = true;
-        return resolve();
+        resolve();
       });
 
       this.producer.on('event.error', (error) => {
         logger.debug(`Error while creating producer: ${error}`, TAG);
-        return reject(error);
+
+        if (reject) {
+          return reject(error);
+        }
+        throw error;
```

Comment:
It seems that throw an error doesn't make sense once nobody will handle it (anonymous function). The reject seems unnecessary too because we set a timeout. I think we could only log the error to debug if something unexpected happened.